### ISSUE TITLE
Support default/base sorting order

### ISSUE THREAD

**Issue by gromakovsky** [serokell/servant-util#22]
If no sorting order is provided to a query, we often still want to apply some _default_ sorting for at least a couple of reasons:
1. To be able to rely on sorting order in tests.
2. To be able to use pagination _safely_. Without any specific order of items pagination doesn't make sense, sometimes there is implicit order, but for example in case of PostgreSQL there seems to be no implicit order, so if you items `0..9` and then `10..19` you may, in theory, get the same items for both queries (since PostgreSQL can order them in any way it wants).

Currently default sorting order can be implemented somehow by user of `servant-util`, but it seems a bit inconvenient and probably should be handled by `servant-util` itself. And note that sometimes you may want the default order to be not available for explicit sorting order. For example, if your items have IDs, you may want to sort them by ID by default (since you have no better option), but at the same time you don't want to make it possible to specify `asc(id)` query because IDs are implementation detail. Also it would be nice if the default order was present in Swagger documentation.

Additionally, even if you have an explicit sorting order, you may still get unpredictable results if there can be multiple items with equal values of the key you are using for sorting.

So, to sum up, I think that at least in case of PostgreSQL backend the ideal sorting approach is to have some default/base sorting order (presumably by primary key) and 1 or more explicit sorting orders. Probably it makes sense to enforce this pattern somehow: either make default sorting order mandatory or somehow make it hard to have no default sorting order.

**Comment by gromakovsky** [comment]
I ended up writing the following function for now:
```
sortingSpecWithId :: forall (params :: [TyNamedParam Type]).
  SortingSpec params -> SortingSpec (("id" ?: SqlSerial Word32) ': params)
sortingSpecWithId = SortingSpec . (<> [sortingItem]) . unSortingSpec
  where
    sortingItem = SortingItem "id" Descendant
```

**Comment by Martoon-00** [comment]
Good observation, I think it's worth implementing.

Your workaround seems proper for now.

I'll probably add another type-level list argument in `SoringParam` and `SortingSpec` that would stand for the base sorting. E.g.

```haskell
SortingParam '[...] '["id" ?: '(SqlSerial Word32, 'Ascending)]
```

And if the field is mentioned in both lists, it would mean that there is a base sorting on it that can be overridden.

I'm not yet sure how should this affect the code in the entrypoints implementation (i.e. what `sortBy_` should accept), probably I'll accept the part respective to the base sorting at the end of the list, all except for the duplicates with the user-specified part.

**Comment by Martoon-00** [comment]
However, I'm really not sure whether will I be able to work on this. If @pasqu4le does not object, I would spend 3-4 hours on this, which should be sufficient.

**Comment by pasqu4le** [comment]
@Martoon-00 no objections from me :+1:

**Comment by Martoon-00** [comment]
:tada: :clap:
